aerogear/mobile-security-service-operator
# AEROGEAR-9985 - Upgrade mss-operator to operator-sdk v0.10.0

Verification instructions:

Enable anyuid:
```
minishift addons enable anyuid
minishift addons apply anyuid
```

Install Integreatly application monitoring operator:
```
clone https://github.com/integr8ly/application-monitoring-operator/ under your GOPATH.
make cluster/install
```

If you see an error in Prometheus regarding file system permissions, you need to `chmod 777` the host path:
```
oc get pvc
oc get pv pv0058 -o yaml
# host path is /var/lib/minishift/base/openshift.local.pv/pv0058
minishift ssh
sudo chmod -R 777 /var/lib/minishift/base/openshift.local.pv/pv0058
exit
# delete prometheus pod to have the statefulset recreate it
```

Deploy MSS operator, monitoring for it, and an example app
```
# first, change image in operator.yaml to aliok/mss_opr_oct_1_1337
make install
make monitoring/install
oc project mobile-security-service-apps
make example-app/apply
```


Check these:
1. MSS is running, you can see the example app in MSS console
2. You can see MSS operator and MSS service in Prometheus targets
3. You can see MSS operator and MSS service related rules in Prometheus rules
4. You can see MSS operator and MSS service related dashboards in Grafana

Note that some of the stuff in Prom/Grafana doesn't show up properly in Minishift. I assume they will show up in RHMI cluster as these Prom queries came all the way until today.
